ledger/participant-integration-api/src/main/scala/platform/apiserver/services/tracking/QueueBackedTracker.scala

```diff
@@ -47,34 +47,23 @@ private[services] final class QueueBackedTracker(
     )
     logger.trace("Tracking command")
     val trackedPromise = Promise[Either[CompletionFailure, CompletionSuccess]]()
-    queue
-      .offer(Ctx(trackedPromise, submission))
-      .flatMap[Either[TrackedCompletionFailure, CompletionSuccess]] {
-        case QueueOfferResult.Enqueued =>
-          trackedPromise.future.map(
-            _.left.map(completionFailure => QueueCompletionFailure(completionFailure))
-          )
-        case QueueOfferResult.Failure(t) =>
-          toQueueSubmitFailure(
-            errorFactories.SubmissionQueueErrors
-              .failedToEnqueueCommandSubmission("Failed to enqueue")(t)
-          )
-        case QueueOfferResult.Dropped =>
-          toQueueSubmitFailure(errorFactories.bufferFull("The submission ingress buffer is full"))
-        case QueueOfferResult.QueueClosed =>
-          toQueueSubmitFailure(
-            errorFactories.SubmissionQueueErrors.queueClosed("Command service queue")
-          )
-      }
-      .recoverWith {
-        case i: IllegalStateException
-            if i.getMessage == "You have to wait for previous offer to be resolved to send another request" =>
-          toQueueSubmitFailure(errorFactories.bufferFull("The submission ingress buffer is full"))
-        case t =>
-          toQueueSubmitFailure(
-            errorFactories.SubmissionQueueErrors.failedToEnqueueCommandSubmission("Failed")(t)
-          )
-      }
+    queue.offer(Ctx(trackedPromise, submission)) match {
```

Comment:
Can this throw an exception? 
We don't handle that, and in case it throws (eg: if the queue is closed maybe?) it would escape the future context and kill us silently

Reply:
I don't think it can. Based on the tests, all the exceptions are being returned as `QueueOfferResult.Failure`.

Reply:
> if the queue is closed maybe

A closed queue has its own result.

Reply:
Ok, I checked the code of the BoundedSourceQueue and I see a throw. I have to revisit this.

Reply:
Ignore, this is not `offer`, so we are fine.